crates/bevy_ecs/src/world/command_queue.rs

```diff
@@ -120,7 +120,32 @@ impl CommandQueue {
         // flush the previously queued entities
         world.flush_entities();
 
+        // Apply any commands in the world's internal queue first.
+        while !world.command_queue.is_empty() {
+            // Put our queue into the `swap_queue`
+            std::mem::swap(self, &mut world.swap_queue);
+            // Take the full queue from the world and replace it with the `swap_queue`
+            std::mem::swap(self, &mut world.command_queue);
+            // Apply the queue
+            self.apply_or_drop_queued(Some(world));
+            // Get our original queue back
+            std::mem::swap(self, &mut world.swap_queue);
+        }
+
+        // Apply our queue
         self.apply_or_drop_queued(Some(world));
+
+        // Apply any resulting commands from the world's internal queue
+        while !world.command_queue.is_empty() {
+            // Put our empty queue into the `swap_queue`
+            std::mem::swap(self, &mut world.swap_queue);
```

Comment:
Functionally this seems correct -- however the `std::mem::swap` between `self` and `world.swap_queue` seems unnecessarily ""clever"". Can you just have a loop that applies the world's command queue, then a loop that applies `self`'s command queue, followed by a loop that applies the world's command queue again if applicable?

Reply:
How do we apply the world's command queue without moving it out of the world?

We could create another command queue on the stack and juggle with that one instead but I don't think that makes anything simpler. Alternatively we can separate out a function that operates on the pointers and pass the pointers to the swap_queue using unsafe without removing it from world but that seems unnecessarily obtuse.

My first instinct was similar to yours but I think this may actually be the simplest implementation from where we stand now.

Reply:
Hmm yeah. I suppose if any commands apply a `CommandQueue` of their own, then the world's `swap_queue` would have to get used.